### PR TITLE
[DataLayout] Extract loop body into a function to reduce nesting (NFC)

### PR DESCRIPTION
**Pull request by s-barannikov**
Also, use `iterator_range` version of `split`.

### CODE REVIEW

**Comment by s-barannikov** [comment]
It is easier viewed with "show whitespace changes" disabled.

File: llvm/lib/IR/DataLayout.cpp [hunk 1]
```diff
@@ -293,289 +293,298 @@ static Error getAddrSpace(StringRef R, unsigned &AddrSpace) {
...
+
+Error DataLayout::parseLayoutString(StringRef LayoutString) {
+  StringRepresentation = std::string(LayoutString);
+
+  if (LayoutString.empty())
+    return Error::success();
+
+  // Split the data layout string into specifications separated by '-'.
+  SmallVector<StringRef, 16> Specs;
+  LayoutString.split(Specs, '-');
```
Comment (nikic) [inline llvm/lib/IR/DataLayout.cpp:580]: Why the upfront split into a vector? There is an iterator version of this (`StringRef Spec : split(LayoutString, '-'))` in StringExtras.h).

Comment (s-barannikov) [reply llvm/lib/IR/DataLayout.cpp]: I just didn't find it, thanks for the suggestion.

**Review by nikic — APPROVED** [review]
LGTM

**Review by bjope — APPROVED** [review]
LG

**Pull request merged**